internal/pkg/composable/providers/kubernetes/hints.go

```diff
@@ -256,19 +256,77 @@ func GetHintsMapping(k8sMapping map[string]interface{}, logger *logp.Logger, pre
 		composableMapping: mapstr.M{},
 		processors:        []mapstr.M{},
 	}
+	var hints mapstr.M
+	var containerProcessors []mapstr.M
 
 	if ann, ok := k8sMapping["annotations"]; ok {
 		annotations, _ := ann.(mapstr.M)
-		hints := utils.GenerateHints(annotations, "", prefix)
+
+		if containerEntries, err := annotations.GetValue(prefix + ".hints"); err == nil {
+			entries, ok := containerEntries.(mapstr.M)
+			if ok && len(entries) > 0 {
+				for key := range entries {
+					parts := strings.Split(key, "/")
+
+					if len(parts) > 1 {
+						if con, ok := k8sMapping["container"]; ok {
+							containers, ok := con.(mapstr.M)
+							if ok {
+								if cname, err := containers.GetValue("name"); err == nil {
+									if parts[0] == cname {
+										// If there are hints like co.elastic.hints.<container_name>/ then add the values after the / to the corresponding container
+										// Eg Annotation "co.elastic.hints.nginx/stream: stderr" will create a hints entry for container nginx
+										hints, containerProcessors = GenerateHintsForContainer(annotations, parts[0], prefix)
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		} else {
+			// If there are top level hints like co.elastic.hints/ then just add the values after the /
+			// Eg. Annotation "co.elastic.hints/stream: stderr" will will create a hints entries for all containers in the pod
+			hints = utils.GenerateHints(annotations, "", prefix)
+		}
```

Comment:
[Suggestion]
This code is already quite "wide", I believe following the convention of asserting `err!=nil`, and therefore swapping the `if` and `else` bodies would make it easier to read.

Reply:
I had a look for this @AndersonQ , thing is that I have this block:
```bash
entries, ok := containerEntries.(mapstr.M)
			if ok && len(entries) > 0 {
```
inside the main if that asserts err==nil and becomes more complex to revert because I would need to make the length check in one command. 

I am thinking of keeping the structure as is for the moment if no big objections